docs/crystal_design.md

```diff
@@ -0,0 +1,187 @@
+# Metadata Management (Crystal)
+
+**Author(s)**: [Pravin Ranjan](https://github.com/PravinRanjan10)
+
+## Motivation and background
+
+Metadata gives basic information about data—including type, time of creation, size of object/buckets, storage class, and more for cloud object storages. Metadata management is the business discipline of managing the metadata about data. It can play key roles during decision making activity. Thus metadata management is required nowadays.
```

Comment:
such as which files in a data lake contain needed data for a particular application or the date of last backup for a backup application